src/libs/elektra/hooks.c

```diff
@@ -81,6 +101,64 @@ static int initHooksSpec (KDB * kdb, Plugin * plugin, Key * errorKey)
 	return 0;
 }
 
+static int initHooksSendNotifications (KDB * kdb, const KeySet * config, KeySet * modules, const KeySet * contract, Key * errorKey)
+{
+	struct _SendNotificationHook * lastHook = kdb->hooks.sendNotification = NULL;
+
+	Key * pluginsKey = keyNew ("system:/elektra/hook/notification/send/plugins", KEY_END);
+	KeySet * configuredPlugins = elektraArrayGet (pluginsKey, config);
```

Comment:
I **really** dislike `elektraArrayGet`. It is a super inefficient way of dealing with arrays. It copies the array part into a separate `KeySet` (which is almost never needed). On top of that it doesn't even use `ksCut` or some other smart way of finding the array. It just iterates _the whole_ `KeySet` and for every key checks if it is an array element.

A better way for iterating the array elements (without `libelektra-ease`) would be:

```c
for (elektraCursor end, it = ksFindHierarchy (config, pluginsKey, &end); it < end; it++)
{
    Key * cur = ksAtCursor (config, it);
    if (keyIsDirectlyBelow (pluginsKey, cur) == 0)
    {
        continue;
    }

    const char * pluginName = keyString (cur);
    // ...
}
```

This doesn't actually check whether the keys are array elements (e.g. `system:/elektra/hook/notification/send/plugins/foo` would work), but we don't really need to check that. If you do want to check, you can use `elektraArrayValidateName (cur) > 0`, but that needs `libelektra-ease` again.